x-pack/plugins/maps/public/shared/layers/sources/es_search_source/create_source_editor.js

```diff
@@ -108,12 +108,16 @@ export class CreateSourceEditor extends Component {
       indexPatternId,
       geoField,
     } = this.state;
+
     if (indexPatternId && geoField) {
       this.props.onSelect({
         indexPatternId,
         geoField,
       });
+      return;
     }
+
+    this.props.onSelect();
```

Comment:
same here wrt implied undefined arg.

Just from a cosmetics POC, the early return here is in the expected code-path. guard statements with early returns are more readable imho when it gets all the oddities out of the way first.

this one in particular, it comes down to the same. I'd consider something more vanilla like.

```
const selection = (indexPatternId && geoField) ? {indexPatternId,geoField} : null;
this.props.onSelect(selection);

```